lib/game.rb

```diff
@@ -0,0 +1,101 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+  WIN_COMBINATIONS = [
+    [0, 1, 2], 
+    [3, 4, 5], 
+    [6, 7, 8], 
+    [0, 3, 6], 
+    [1, 4, 7], 
+    [2, 5, 8], 
+    [0, 4, 8], 
+    [2, 4, 6]
+    ]
+    
+    def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+      @player_1 = player_1
+      @player_2 = player_2
+      @board = board
+    end
+    
+    def current_player
+      if board.turn_count % 2 == 0
+        player_1
+      else
+        player_2
+      end
+    end
+    
+    def won?
+      WIN_COMBINATIONS.each do |winning_line|
+        if board.cells[winning_line[0]] == "X" && board.cells[winning_line[1]] == "X" && board.cells[winning_line[2]] == "X"
+          return [winning_line[0], winning_line[1], winning_line[2]]
+          elsif board.cells[winning_line[0]] == "O" && board.cells[winning_line[1]] == "O" && board.cells[winning_line[2]] == "O"
+          return [winning_line[0], winning_line[1], winning_line[2]]
+        end
+      end
+      false
+    end
+    
+    def draw?
+      board.full? && !won?
+    end
+    
+    def over?
+      draw? || won?
+    end
+    
+    def winner
+      if won?
+        board.cells[won?[0]]
+      end
+    end
+    
+    def turn
+      player = current_player
+      current_move = player.move(board)
+      if !board.valid_move?(current_move)
+        turn
+      else
+        puts "Turn: #{@board.turn_count+1} || Player: #{player.token}"
+        puts "\n"
+        @board.update(current_move, player)
+        puts "#{player.token} moved to space #{current_move}."
+        @board.display
+      end
+    end
+    
+    def play
+      while !over?
+        turn
+      end
+      
+      if won?
+        puts "Congratulations #{winner}!"
+      else
+        puts "Cat's Game!"
+      end
+    end
+    
+  def start_game
```

Comment:
We would prefer to see the CLI logic separated from the game logic in its own class (e.g. TicTacToeCLI). Splitting the interface from the implementation of the game allows for multiple interfaces (e.g. a web app) without requiring changes to the game code.

So for example, you could encapsulate CLI logic into a class and the only code in your bin executable would be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.start_game
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def start_game
   # Your game logic
  end
end
```